manifests/suse.pp

```diff
@@ -13,13 +13,23 @@
 ) inherits datadog_agent::params {
 
   $current_key = 'https://keys.datadoghq.com/DATADOG_RPM_KEY_CURRENT.public'
-  $all_keys = [
+  $keys_src = [
     $current_key,
-    'https://keys.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public',
     'https://keys.datadoghq.com/DATADOG_RPM_KEY_FD4BF915.public',
     'https://keys.datadoghq.com/DATADOG_RPM_KEY_B01082D3.public',
+    'https://keys.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public',
   ]
 
+  if $agent_version =~ /([0-9]+:)?([0-9]+)\.([0-9]+)\.([0-9]+)((?:~|-)[^0-9\s-]+[^-\s]*)?(?:-([0-9]+))?/ or $agent_version == 'latest' {
+      if $agent_major_version > 5 and ($agent_version == 'latest' or 0 + $3 > 35) {
+        $all_keys = $keys_src[0,3]
+      } else {
+        $all_keys = $keys_src
+      }
+  } else {
+    $all_keys = $keys_src
```

Comment:
nit: it seems `keys_src` potentially contains more keys than `all_keys`. I suppose it would be more readable to invert the variables